src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -140,7 +140,7 @@ public String getScript() {
         FilePath resultFile = c.getResultFile(ws);
         FilePath controlDir = c.controlDir(ws);
         if (capturingOutput) {
-            cmd = String.format("{ while [ -d '%s' -a \\! -f '%s' ]; do touch '%s'; sleep 3; done } & jsc=%s; %s=$jsc '%s' > '%s' 2> '%s'; echo $? > '%s.tmp'; mv '%s.tmp' '%s'; wait",
+            cmd = String.format("pid=$$; { while ps -o pid -p $pid | grep -q $pid && [ -d '%s' -a \\! -f '%s' ]; do touch '%s'; sleep 3; done } & jsc=%s; %s=$jsc '%s' > '%s' 2> '%s'; echo $? > '%s.tmp'; mv '%s.tmp' '%s'; wait",
```

Comment:
Would it be better to use `grep -q "^$pid$"` in case the pid is a substring of some other valid pid on the agent? (Same for below as well)

Reply:
This apparently just caused a problem for me in the wild:

```
15:25:26 ps: unrecognized option: p
15:25:26 BusyBox v1.27.2 (2017-12-12 10:41:50 GMT) multi-call binary.
15:25:26 
15:25:26 Usage: ps [-o COL1,COL2=HEADER]
15:25:26 
15:25:26 Show list of processes
15:25:26 
15:25:26 	-o COL1,COL2=HEADER	Select columns for display
```

Had to revert to the previous version.

Reply:
@oeuftete Best to open a bug report on https://issues.jenkins-ci.org/secure/Dashboard.jspa. It would be great if you could include the output of  `ps -o pid`, `ps -o pid=`, and `ps -o pid=HEADER` on your system.

Reply:
@oeuftete Also, please include your operating system and version for the **build agent** in that ticket.

Reply:
Looks like [BusyBox's ps](https://busybox.net/downloads/BusyBox.html?#ps) only has `-o` and `-T` options, so we'd have to omit the `-p` option and pass a stricter pattern to `grep` for this to work with that version of `ps`.

Notably, Alpine Linux's Official Docker image only has BusyBox's `ps` installed by default, see gliderlabs/docker-alpine#173, so we probably want to fix it.

Reply:
jenkins-slave image is adding `procps` in https://github.com/jenkinsci/docker-slave/pull/28

Reply:
https://issues.jenkins-ci.org/browse/JENKINS-52847